CMakeLists.txt

```diff
@@ -106,25 +97,39 @@ else()
 endif()
 
 if (ADDRESS_SANITIZER_ENABLED)
-    set(CMAKE_NO_BUILTIN_CHRPATH ON)
-    rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
+    rocm_check_target_ids(DEFAULT_GPU_TARGETS
         TARGETS "gfx90a:xnack+;gfx942:xnack+" )
 else()
-    rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
+    rocm_check_target_ids(DEFAULT_GPU_TARGETS
         TARGETS "gfx908;gfx90a;gfx942" )
 endif()
 
-# Variable AMDGPU_TARGET must be a cached variable and must be specified before calling find_package(hip)
-# This is because hip-config.cmake sets --offload-arch via AMDGPU_TARGET cached variable __after__ setting
-# default cached variable AMDGPU_TARGET to "gfx900;gfx906;gfx908;gfx1100;gfx1101;gfx1102", where not all archs are compatible with MFMA instructions
+# Check if offload compression is supported
+include(CheckCXXCompilerFlag)
+if (BUILD_OFFLOAD_COMPRESS)
```

Comment:
I think BUILD_OFFLOAD_COMPRESS and CMAKE_NO_BUILTIN_CHRPATH should be mutually exclusive. Always try to reduce the size with offload compression. If that's not available, we need to provide the linking workaround for no_builtin_chrpath for large binaries.

Something like this: 
`
if (BUILD_OFFLOAD_COMPRESS)
{

    // mutual exclusion
    if ( no compiler support) enable CMAKE_NO_BUILTIN_CHRPATH
    else disable CMAKE_NO_BUILTIN_CHRPATH
}
else enable CMAKE_NO_BUILTIN_CHRPATH // no compression, expect large binaries. Not dependent on asan
`

Reply:
@CongMa13 can you confirm asan dependency on CMAKE_NO_BUILTIN_CHRPATH?

Reply:
yeah, `CMAKE_NO_BUILTIN_CHRPATH` was added since a CMake bug that it cannot handle ELF files with more than 65536 sections. The hiptensor library was greater than 2GB at that time. 

With the less targets and the compress flag, the file size is only 174M now. So it is safe to remove CMAKE_NO_BUILTIN_CHRPATH.